core/events/ui_events.js

```diff
@@ -11,42 +11,73 @@
 'use strict';
 
 goog.provide('Blockly.Events.Ui');
+goog.provide('Blockly.Events.UiBase');
 
 goog.require('Blockly.Events');
 goog.require('Blockly.Events.Abstract');
 goog.require('Blockly.registry');
 goog.require('Blockly.utils.object');
 
-
 /**
- * Class for a UI event.
+ * Base class for a UI event.
  * UI events are events that don't need to be sent over the wire for multi-user
  * editing to work (e.g. scrolling the workspace, zooming, opening toolbox
  * categories).
  * UI events do not undo or redo.
+ * @param {string=} opt_workspaceId The workspace identifier for this event.
+ *    Undefined for a blank event.
+ * @extends {Blockly.Events.Abstract}
+ * @constructor
+ */
+Blockly.Events.UiBase = function(opt_workspaceId) {
+  Blockly.Events.UiBase.superClass_.constructor.call(this);
+
+  /**
+   * Whether or not the event is blank (to be populated by fromJson).
+   * @type {boolean}
+   */
+  this.isBlank = typeof opt_workspaceId == 'undefined';
+
+  /**
+   * The workspace identifier for this event.
+   * @type {string}
+   */
+  this.workspaceId = opt_workspaceId ? opt_workspaceId : '';
+
+  // UI events do not undo or redo.
+  this.recordUndo = false;
+};
+Blockly.utils.object.inherits(Blockly.Events.UiBase, Blockly.Events.Abstract);
+
+/**
+ * Whether or not the event is a UI event.
+ * @type {boolean}
+ */
+Blockly.Events.UiBase.prototype.IS_UI_EVENT = true;
+
+/**
+ * Class for a UI event.
  * @param {?Blockly.Block=} opt_block The affected block.  Null for UI events
  *     that do not have an associated block.  Undefined for a blank event.
  * @param {string=} opt_element One of 'selected', 'comment', 'mutatorOpen',
  *     etc.
  * @param {*=} opt_oldValue Previous value of element.
  * @param {*=} opt_newValue New value of element.
- * @extends {Blockly.Events.Abstract}
+ * @extends {Blockly.Events.UiBase}
+ * @deprecated Instead use a more specific UI event.
```

Comment:
Add when it was deprecated so we know approximately when we should remove.

Reply:
Done.